src/libpriv/rpmostree-core.c

```diff
@@ -1954,7 +1956,7 @@ rpmostree_context_prepare (RpmOstreeContext *self,
   g_autoptr(GPtrArray) missing_pkgs = NULL;
   for (char **it = pkgnames; it && *it; it++)
     {
-      const char *pkgname = *it;
+      const char *pkgname = g_hash_table_lookup (self->vlockmap, *it) ?: *it;
```

Comment:
The issue with this is that the manifest might not specify just a package name. It might be e.g. `foo-1.4` or `something-that-foo-provides`. I guess we could just *require* that manifests be in pure "pkgname" format only when using lockfiles.

Though one thing we could also do here which would probably be more explicit anyway is, for each `pkgname`, do `rpmostree_get_matching_packages` (this is similar to what `dnf_context_install` does), then iterate through the list of matching NEVRAs and check if there's one in the lockfile. If so, then verify the `DnfPackage` checksum matches and `hy_goal_install()` on that one. Otherwise, just pass the pkgname directly to `dnf_context_install` as we do here when it's not in the hash table.

Reply:
Resolved.